app/jinja_filters.py

```diff
@@ -247,66 +247,38 @@ def __init__(self, _for, text, description=None):
         self.description = description
 
 
-class CheckboxConfig:
-    def __init__(self, option, index, form, answer):
+class SelectConfig:
+    def __init__(self, option, index, answer, form=None):
         self.id = option.id
         self.name = option.name
         self.value = option.data
         self.checked = option.checked
 
         label_description = None
-        answer_option = answer["options"][index]
 
-        if answer_option and "description" in answer_option:
-            label_description = answer_option["description"]
+        try:
+            self._answer_option = answer.get("options", [])[index]
```

Comment:
This is interesting. I might need an example, is it trying to do 2 things at once. 
1. If options doesn't exist if defaults to an empty list which will fall over because it has no index 
2. If there are options, but an index is supplied which doesn't exist it also falls over

Also interested why we then assign an empty dict

Reply:
In either case, we want to get the item based on an index, which won't exist for dynamic answers. 

The empty dict assignment is so that the type of the variable is a single type (dict) but nothing stopping us making it an optional dict and assigning None. I'm easy either way.